microhapulator/Snakefile

```diff
@@ -0,0 +1,369 @@
+        with open(templatefile, "r") as infh, open("report.html", "w") as outfh:
+            template = Template(infh.read())
+            output = template.render(
+                date=datetime.now().replace(microsecond=0).isoformat(),
+                mhpl8rversion=microhapulator.__version__,
+                samples=config["samples"],
+                summary=summary,
+                r1readlenplots=r1readlenplots,
+                r2readlenplots=r2readlenplots,
+                mergedreadlenplots=mergedreadlenplots,
+                locbalanceplots=locbalanceplots,
+                hetbalanceplots=hetbalanceplots,
+                static=5,
+                dynamic=0.02,
+                zip=zip,
+            )
+            print(output, file=outfh, end="")
+
+
+rule summary:
+    input:
+        expand("analysis/{sample}/flash.log", sample=config["samples"]),
+        expand("analysis/{sample}/{sample}-mapped-reads.txt", sample=config["samples"]),
+        expand(
+            "analysis/{sample}/fullrefr/{sample}-fullrefr-mapped-reads.txt",
+            sample=config["samples"],
+        ),
+        expand("analysis/{sample}/{sample}-typing-rate.tsv", sample=config["samples"]),
+        expand("analysis/{sample}/{sample}-interlocus-balance-chisq.txt", sample=config["samples"]),
+        expand(
+            "analysis/{sample}/{sample}-heterozygote-balance-pttest.txt",
+            sample=config["samples"],
+        ),
+    output:
+        tsv="analysis/summary.tsv",
+    run:
+        columns = (
+            "Sample",
+            "TotalReads",
+            "Merged",
+            "MergeRate",
+            "Mapped",
+            "MappingRate",
+            "MappedFullRefr",
+            "MappingRateFullRefr",
+            "Typed",
+            "TypingRate",
+            "InterlocChiSq",
+            "HetTstat",
+        )
+        data = {colname: [] for colname in columns}
+        for sample in sorted(config["samples"]):
+            print(f"[Compiling summary] Sample={sample}", file=sys.stderr)
+            totalreads, mergedreads, *_ = parse_flash_summary(f"analysis/{sample}/flash.log")
+            maptotal, mapped = parse_read_counts(f"analysis/{sample}/{sample}-mapped-reads.txt")
+            assert maptotal == mergedreads, (sample, maptotal, mergedreads)
+            frmaptotal, frmapped = parse_read_counts(
+        f"analysis/{sample}/fullrefr/{sample}-fullrefr-mapped-reads.txt"
+            )
+            assert (
+                frmaptotal == maptotal
+            ), f"Sample={sample} fullrefr map total={frmaptotal} map total={maptotal}"
+            typing = pd.read_csv(f"analysis/{sample}/{sample}-typing-rate.tsv", sep="\t")
+            num_typed_reads = typing.TypedReads.sum()
+            typing_total_reads = typing.TotalReads.sum()
+            # assert typing_total_reads == mapped, f"Sample={sample} type total={typing_total_reads} mapped={mapped}"
```

Comment:
This should be removed if it is no longer being used.

Reply:
Comment added in 92552c2775fc72c4a84555b7d309082ecb1bd704.